pkg/model/requests_processing.go

```diff
@@ -56,21 +58,49 @@ func (rps *requestsProcessingStock) EntitiesInStock() []*simulator.Entity {
 }
 
 func (rps *requestsProcessingStock) Remove() simulator.Entity {
-	return rps.delegate.Remove()
+	request := rps.delegate.Remove().(*requestEntity)
+	*rps.occupiedCPUCapacityMillis -= *request.utilizationForRequest
+	return request
 }
 
 func (rps *requestsProcessingStock) Add(entity simulator.Entity) error {
 	rps.numRequestsSinceLast++
+	request := *entity.(*requestEntity)
+
+	freeCPUCapacityMillis := *rps.totalCPUCapacityMillis - *rps.occupiedCPUCapacityMillis
+	if freeCPUCapacityMillis == 0 {
+		freeCPUCapacityMillis = 1
+	}
+	cpuTime := request.requestConfig.CPUTimeMillis * 1000 / freeCPUCapacityMillis
```

Comment:
We seem to assume that the request will run with the capacity that it saw in the beginning, don't we? But in reality, when other requests finished and freed CPU, the request would speed up. Are we ignoring it?

Reply:
I ignored it in the original version and relied on the Sakasegawa approximation. I'd put off a deeper simulation like this one as a placeholder, because I felt it would become a serious rabbit hole. There's a queueing theory finding called the "arrival theorem", loosely that an arrival sees the same time averages as an outside observer would. In that sense the figure we calculate at arrival time encodes the future evolution of the queue.

But some day I would definitely like to increase model fidelity. I don't feel mathematically mature enough to state confidently whether the arrival theorem applies in this case.